### PR TITLE
Add e3nn scorer

### LINKED ISSUE #839: Implement E3NN Scorer
We need to add in an E3NN scorer (basically same API as Schnet scorer)

### PR DESCRIPTION
**Pull request by hmacdope**
## Description
Provide a brief description of the PR's purpose here.
Fixes #839 

Branches off #810 so that must go in first.  

## Developers certificate of origin
- [X] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by hmacdope** [comment]
@kaminow would you be able to have a look at this error I am getting from MTENN internals?


```python

    def forward(self, data: Union[Data, Dict[str, torch.Tensor]]) -> torch.Tensor:
        """evaluate the network
    
        Parameters
        ----------
        data : `torch_geometric.data.Data` or dict
            data object containing
            - ``pos`` the position of the nodes (atoms)
            - ``x`` the input features of the nodes, optional
            - ``z`` the attributes of the nodes, for instance the atom type, optional
            - ``batch`` the graph to which the node belong, optional
        """
        if "batch" in data:
            batch = data["batch"]
        else:
            batch = data["pos"].new_zeros(data["pos"].shape[0], dtype=torch.long)
    
        edge_index = radius_graph(data["pos"], self.max_radius, batch)
        edge_src = edge_index[0]
        edge_dst = edge_index[1]
        edge_vec = data["pos"][edge_src] - data["pos"][edge_dst]
        edge_sh = o3.spherical_harmonics(self.irreps_edge_attr, edge_vec, True, normalization="component")
        edge_length = edge_vec.norm(dim=1)
        edge_length_embedded = soft_one_hot_linspace(
            x=edge_length, start=0.0, end=self.max_radius, number=self.number_of_basis, basis="gaussian", cutoff=False
        ).mul(self.number_of_basis**0.5)
        edge_attr = smooth_cutoff(edge_length / self.max_radius)[:, None] * edge_sh
    
        if self.input_has_node_in and "x" in data:
            assert self.irreps_in is not None
            x = data["x"]
        else:
>           assert self.irreps_in is None
E           AssertionError
```

**Comment by kaminow** [comment]
this is just an issue of the different models expecting different names in the `pose` dict. should be fixed in my most recent commit, but feel free to rearrange the code to make it nicer. in the future, we should be able to avoid stuff like this once choderalab/mtenn#39 is resolved

**Comment by hmacdope** [comment]
@kaminow can I grab a review from you here?

File: asapdiscovery-docking/asapdiscovery/docking/scorer.py [hunk 1]
```diff
@@ -351,6 +352,27 @@ def _score(self, inputs: list[DockingResult]) -> list[Score]:
...
+        for inp in inputs:
+            e3nn_score = self.inference_cls.predict_from_oemol(inp.to_posed_oemol())
+            results.append(
+                Score.from_score_and_docking_result(
+                    e3nn_score, self.score_type, self.units, inp
+                )
+            )
+        return results
+
+
```
Comment (kaminow) [inline asapdiscovery-docking/asapdiscovery/docking/scorer.py:375]: should we add `E3NNScorer` to the list below?

File: asapdiscovery-docking/asapdiscovery/docking/scorer.py [hunk 2]
```diff
@@ -351,6 +352,27 @@ def _score(self, inputs: list[DockingResult]) -> list[Score]:
         return results
 
 
+class E3NNScorer(MLModelScorer):
+    """
+    Scoring using Schnet ML Model
```
Comment (kaminow) [inline asapdiscovery-docking/asapdiscovery/docking/scorer.py:357]:
```suggestion
    Scoring using e3nn ML Model
```

**Review by kaminow — APPROVED** [review]
some super minor changes but otherwise looks good!

**Comment by hmacdope** [comment]
@kaminow thanks! All fixed up, i'll merge now.

**Pull request merged**